### PR TITLE
Fixes

### LINKED ISSUE #78: MultiDashboard::save() overwrites attribute values
I am creating a MultiDashboard instance programmatically and save() keeps overwriting my owner and container values.

### LINKED ISSUE #80: Only show tab settings icon if can edit
Tab settings icon is visible even if user can't edit the tab

### LINKED ISSUE #82: Delete fixed widgets from dashboards when parent is deleted
When parent widget is deleted from default widgets interface all fixed widgets from all dashboards should be removed. 
You can steal the code from here:
https://github.com/hypeJunction/Elgg-roles_dashboard/blob/master/start.php#L481

### LINKED ISSUE #83: MultiDashboard::getURL() doesn't trigger hooks
The dashboard URL is set in stone and is impossible to modify without overriding the class

### PR DESCRIPTION
**Pull request by hypeJunction**
**fix(classes): MultiDashboard::save() no longer sets attributes**
`MultiDashboard::save()` was duplicating logic done in `ElggObject` and `ElggEntity`, additionally creating problems when dashboards were created programmatically.
Fixes #78

**feature(url): dashboard entity URLs can now be filtered**
Dashboard URL can now be filtered by core `entity:url` hook.
Fixes #83

**fix(ui): only show edit icon if dashboard is editable**
Fixes #80

### CODE REVIEW

**Comment by hypeJunction** [comment]
You may need to look into your Travis integration - composer seemed to have failed to resolve dependencies.

File: lib/events.php [hunk 1]
```diff
@@ -225,3 +225,38 @@ function widget_manager_update_group_event_handler($event, $object_type, $object
...
+function widget_manager_remove_fixed_widgets($event, $type, $entity) {
+	if (!$entity instanceof ElggWidget) {
+		return;
+	}
+	if (!stristr($_SERVER["HTTP_REFERER"], "/admin/appearance/default_widgets")) {
+		// We only care about widgets that were deleted from default_widgets interface
+		return;
+	}
+
+	$fixed_widgets = new ElggBatch('elgg_get_entities_from_private_settings', array(
```
Comment (jdalsem) [inline lib/events.php]: i think you need batch->setIncrementOffset(false) if you are deleting

Comment (hypeJunction) [inline lib/events.php]: Indeed

Comment (hypeJunction) [inline lib/events.php]: Done.

File: lib/events.php [hunk 2]
```diff
@@ -225,3 +225,40 @@ function widget_manager_update_group_event_handler($event, $object_type, $object
...
+/**
+ * Remove fixed widgets from dashboard when parent widget is deleted from
+ * default widgets editor
+ *
+ * @param string     $event  "delete"
+ * @param string     $type   "object"
+ * @param ElggWidget $entity Widget
+ * @return void
+ */
+function widget_manager_remove_fixed_widgets($event, $type, $entity) {
```
Comment (jdalsem) [inline lib/events.php]: Do we really want to delete the fixed widgets? Why not 'release' them? That way users can keep using the widget and only remove if they no longer need it

Comment (hypeJunction) [inline lib/events.php]: I guess it really depends on the use case. Theoretically, users can just add them if they want to again. In my project, users can't edit the layout and all widgets are kept in sync. Up to you really.

Comment (jdalsem) [inline lib/events.php]: i think the more common use case is to release the widgets (but i believe that is currently already happening), so deleting them is something you could easily do in your own use case...

Comment (hypeJunction) [inline lib/events.php]: I removed that commit.

**Pull request merged**